pkg/signer/preconfsigner/signer.go

```diff
@@ -272,6 +284,8 @@ func GetPreConfirmationHash(c *PreConfirmation) ([]byte, error) {
 	data := append(eip712MessageTypeHash.Bytes(), txnHashHash.Bytes()...)
 	data = append(data, math.U256Bytes(c.Bid.BidAmt)...)
 	data = append(data, math.U256Bytes(c.Bid.BlockNumber)...)
+	data = append(data, math.U256Bytes(big.NewInt(int64(c.Bid.DecayStartTimeStamp)))...)
```

Comment:
I suggest setting the Uint64 value directly, instead of casting it to int64. This approach avoids potential overflow issues and makes the intention clearer.
```suggestion
	data = append(data, math.U256Bytes(new(big.Int).SetUint64(c.Bid.DecayStartTimeStamp))...)
```